src/lib/components/drip-list-badge/drip-list-badge.svelte

```diff
@@ -3,53 +3,51 @@
   import ensStore from '$lib/stores/ens';
   import formatAddress from '$lib/utils/format-address';
 
-  export let listName: string;
   export let listId: string;
-  export let owner: string;
+  export let listName: string | undefined = undefined;
+  export let owner: string | undefined = undefined;
 
-  export let showAvatar = true;
-  export let showName = true;
   export let isLinked = true;
+  export let showAvatar = true;
+  export let avatarSize: 'small' | 'default' = 'default';
 
-  $: ensStore.connected && ensStore.lookup(owner);
-  $: ens = $ensStore[owner];
-
-  $: username = ens?.name ? ens.name : formatAddress(owner);
+  // lookup ens name if owner is provided
+  $: owner && ensStore.connected && ensStore.lookup(owner);
+  $: ens = owner ? $ensStore[owner] : {};
+  $: username = ens?.name || (owner && formatAddress(owner));
 </script>
 
 <svelte:element
   this={isLinked ? 'a' : 'div'}
   href={isLinked ? `/app/drip-lists/${listId}` : undefined}
   tabindex={isLinked ? 0 : -1}
-  class="drip-list-badge flex gap-2 items-center"
+  class="drip-list-badge outline-none flex gap-2 items-center"
 >
   {#if showAvatar}
-    <div class="drip-list-icon">
-      <DripListIcon style="fill: var(--color-primary)" />
+    <div
+      class="flex items-center justify-center rounded-full flex-shrink-0 bg-primary-level-1 {avatarSize ===
+      'small'
+        ? 'w-6 h-6'
+        : 'w-8 h-8'}"
+    >
+      <DripListIcon
+        style="fill: var(--color-primary); {avatarSize === 'small'
+          ? 'width:18px; height:18px;'
+          : ''}"
+      />
     </div>
   {/if}
-  {#if showName}
+  {#if listName}
     <div class="name typo-text text-foreground flex-1 min-w-0 truncate">
-      <span><span class="text-foreground-level-5">{username}/</span>{listName}</span>
+      <span
+        >{#if username}<span class="text-foreground-level-5">{username}/</span
+          >{/if}{@html listName}</span
```

Comment:
This seems to open up a XSS vulnerability… could call my list `<script>doBadStuff()</script>` 🥴 

maybe `listName` could be `string | null | undefined` with a default of `null`, where `null` means "no name", and `undefined` means "name is loading"

Reply:
oddly, svelte transforms undefined to null inside the component!
```
listName={dripList ? dripList.name : undefined}

```
outputs null inside the component...
maybe they consider it an error?

Reply:
... but tbh, maybe just a global "loading" prop is way easier and more explicit than any of that 😄